mempool/mempool.go

```diff
@@ -191,3 +279,29 @@ func (e ErrPreCheck) Error() string {
 func IsPreCheckError(err error) bool {
 	return errors.As(err, &ErrPreCheck{})
 }
+
+// MempoolTx is a transaction that successfully ran
+type SidecarTx struct {
+	DesiredHeight int64 // height that this tx wants to be included in
+	BundleID      int64 // ordered id of bundle
+	BundleOrder   int64 // order of tx within bundle
+	BundleSize    int64 // total size of bundle
+
+	GasWanted int64    // amount of gas this tx states it will require
+	Tx        types.Tx // tx bytes
+
+	// ids of peers who've sent us this tx (as a map for quick lookups).
+	// senders: PeerID -> bool
+	Senders sync.Map
```

Comment:
What is this for? I am checking with CometBFT people.

Reply:
AFAIK, this map is used in broadcastSidecarTxRoutine - we do not wish to resend the tx to the sidecar peers we received them from.
The same sync.Map exists for the mempolTx, and the concept is the same within the broadcastMempoolTxRoutine.

Reply:
It makes sense, thanks. Clarify anyway that this is for sentry nodes because there is no assumption about the infrastructure topology.